ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -183,9 +183,10 @@ instance
       -- For evidence of this behaviour, replace the cased-on expression by:
       -- > @traceShowId $ shelleyTriggerHardFork pcf@
       case shelleyTriggerHardFork pcfg of
-        TriggerHardForkNever                         -> Nothing
-        TriggerHardForkAtEpoch   epoch               -> Just epoch
-        TriggerHardForkAtVersion shelleyMajorVersion ->
+        TriggerHardForkNever                         -> FixedTransition Nothing
```

Comment:
Bah! This line would reintroduce the bug fixed by https://github.com/input-output-hk/ouroboros-network/pull/3754

For example, I think `TriggerHardForkNever` should be renamed to `TriggerHardForkCannotBeKnown`.

Either that, or `protocolInfoCardano` should use `TriggerHardForkAtVersion` ever for the last era. 🤔

Reply:
I think PR https://github.com/input-output-hk/ouroboros-network/pull/3754 should have also changed the trigger for `Babbage` to not be `TriggerHardForkNever` 🤦